torch/nn/modules/_functions.py

```diff
@@ -7,7 +7,12 @@ class SyncBatchNorm(Function):
 
     @staticmethod
     def forward(self, input, weight, bias, running_mean, running_var, eps, momentum, process_group, world_size):
-        input = input.contiguous()
+        if not input.is_contiguous(memory_format=torch.channels_last):
+            input = input.contiguous()
```

Comment:
I don't see anywhere where it's checked that weight is contiguous - is there such a place?

Reply:
I think `weight` is initialized as a 1d tensor. Do we need to add a `weight = weight.contiguous()` here?

https://github.com/pytorch/pytorch/blob/bf4fcab68104b333c81e39d142dccc7ae7a6f1d7/torch/nn/modules/batchnorm.py#L40

Reply:
Even if `weight` is initialized as contiguous, there's no guarantee it's not changed later.

Reply:
The call to contiguous should already be a no-op if the Tensor is already contiguous. Maybe you want to pass the memory_format arg directly to the contiguous call?

Reply:
IIUC, the logic here is that if input is not channels-last contiguous, regular nchw path will be taken.